modules/uncertainty/R/prep.data.assim.R

```diff
@@ -0,0 +1,82 @@
+##'@title prep.data.assim
+##'@section purpose:
+##'This function sets up a call to PEcAn.assim.sequential::sda.enkf()
+##'
+##'@param settings the PEcAn settings object (a collection of nested lists)
+##'@return None
+##'@export
+##'@author Luke Dramko
+prep.data.assim <- function(settings) {
+  # Obtain real data from the site
+  timestep <- 0.5 # Every half hour = 0.5
+  
+  field_data <- PEcAn.data.atmosphere::download.US_WCr(settings$run$start_date, settings$run$end_date, timestep = timestep)
+  
+  uncertainty_vals <- list()
+  
+  # Creates a proxy row for rbinding
+  sums <- NULL
+  
+  # One vector holds the mean for each variable.
+  obs.mean <- NULL
+  
+  for (i in 1:length(field_data)) {
+    AMF.params <- PEcAn.uncertainty::flux.uncertainty(field_data[[i]], QC = rep(0, length(field_data[[i]])))
+    numvals <- 10;
+    
+    # Create proxy row for rbinding
+    random_mat = NULL
+    new_col = rep(0, length(field_data[[i]]))
+    
+    # Create a new column
+    # i: the particular variable being worked with
+    # j: the column number
+    # k: the row number
+    for (j in 1:numvals) { # number of random numbers
+      obs <- field_data[[i]][!is.na(field_data[[i]])]
+      plot(obs)
```

Comment:
probably don't want to make a plot during a workflow run, and definitely not 5000 times in a loop